.github/workflows/setup-binary-test.yml

```diff
@@ -0,0 +1,93 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'setup-binary-test'
+
+on:
+  pull_request:
+    paths:
+      - '.github/actions/setup-binary/**'
+jobs:
+  setup-binary-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - 'ubuntu-latest'
+          - 'ubuntu-20.04'
+          - 'windows-latest'
+          - 'windows-2019'
+          - 'macos-latest'
+          - 'macos-11'
+        include:
+          - os: 'ubuntu-latest'
+            download_url: 'https://github.com/abcxyz/abc/releases/download/v0.0.1-alpha4/abc_0.0.1-alpha4_linux_amd64.tar.gz'
+
+          - os: 'ubuntu-20.04'
+            download_url: 'https://github.com/abcxyz/abc/releases/download/v0.0.1-alpha4/abc_0.0.1-alpha4_linux_amd64.tar.gz'
+
+          - os: 'windows-latest'
+            download_url: 'https://github.com/abcxyz/abc/releases/download/v0.0.1-alpha4/abc_0.0.1-alpha4_windows_amd64.zip'
+
+          - os: 'windows-2019'
+            download_url: 'https://github.com/abcxyz/abc/releases/download/v0.0.1-alpha4/abc_0.0.1-alpha4_windows_amd64.zip'
+
+          - os: 'macos-latest'
+            download_url: 'https://github.com/abcxyz/abc/releases/download/v0.0.1-alpha4/abc_0.0.1-alpha4_darwin_amd64.tar.gz'
+
+          - os: 'macos-11'
+            download_url: 'https://github.com/abcxyz/abc/releases/download/v0.0.1-alpha4/abc_0.0.1-alpha4_darwin_amd64.tar.gz'
```

Comment:
Can't we do string interpolation on these based on `runner.os`?

Reply:
We would need to map them...this just seemed eaiser.

macos -> darwin
linux -> ubuntu

Reply:
Right:

```yaml
matrix:
  include:
    - os: 'ubuntu-latest'
      name: 'linux'
    - os: 'windows-latest'
      name: 'windows'
    - os: 'macos-latest'
      name: 'darwin'
```

I'm also not sure why we want to test older versions of Ubuntu or OSX - we should just not support them imo.

Reply:
SGTM